test/ConfigResourceFactoryTest.php

```diff
@@ -32,44 +32,54 @@ class ConfigResourceFactoryTest extends TestCase
     private $factory;
 
     /**
-     * @var WriterInterface
+     * @var WriterInterface|MockObject
+     * @psalm-var WriterInterface&MockObject
      */
     private $writer;
 
     protected function setUp(): void
     {
-        $this->writer = $this->prophesize(WriterInterface::class)->reveal();
-        $this->container = $this->prophesize(ContainerInterface::class);
-        $this->container->get(ConfigWriter::class)->willReturn($this->writer);
+        $this->writer = $this->createMock(WriterInterface::class);
+        $this->container = $this->createMock(ContainerInterface::class);
         $this->factory = new ConfigResourceFactory();
     }
 
-    public function testReturnsInstanceOfConfigResource()
+    public function testReturnsInstanceOfConfigResource(): void
     {
-        $this->container->has('config')->willReturn(false);
+        $this->container->expects($this->atLeastOnce())->method('has')->with('config')->willReturn(false);
+        $this->container
+            ->expects($this->once())
+            ->method('get')
+            ->with(self::WRITER_SERVICE)
+            ->willReturn($this->writer);
 
         $factory = $this->factory;
-        $configResource = $factory($this->container->reveal());
+        $configResource = $factory($this->container);
 
         $this->assertInstanceOf(ConfigResource::class, $configResource);
     }
 
-    public function testDefaultAttributesValues()
+    public function testDefaultAttributesValues(): void
     {
-        $this->container->has('config')->willReturn(false);
+        $this->container->expects($this->atLeastOnce())->method('has')->with('config')->willReturn(false);
+        $this->container
+            ->expects($this->once())
```

Comment:
Once is a static method (at least in the recent phpunit versions) 

```suggestion
            ->expects(self::once())
```